### ISSUE TITLE
bug: `llvm-header-guard` emits wrong warning in case of `-std=c++20`

### ISSUE THREAD

**Issue by thorsten-klein** [llvm/llvm-project#180155]
Header File `include/test.hpp`:
```
#ifndef TEST_HPP
#define TEST_HPP
#endif // TEST_HPP
```

Run:
`clang-tidy '--checks=-*,llvm-header-guard' include/test.hpp -- -std=c++20`

Emits warning:
```
1 warning generated.
/home/user/work/include/test.hpp:2:9: warning: code/includes outside of area guarded by header guard; consider moving it [llvm-header-guard]
    2 | #define TEST_HPP
      |   
```

**Comment by vbvictor** [comment]
This is probably a regression after https://github.com/llvm/llvm-project/pull/143898 / https://github.com/llvm/llvm-project/pull/145244

Confirmed that this is not happening in `clang-tidy-20` and started to fire in `clang-tidy-21` (which was released after those patches were landed)
